syntax = "proto3";

package meterusage;

// MeterUsage server configuration
service MeterUsage {
    // Return usage data
    rpc GetUsageData (UsageRequest) returns (UsageReply) {}
}
  
// The usage request 
message UsageRequest {
    // Name of requested usage statistics
    string usage_stat_name = 1;
    // Number of page
    int32 page_number = 2;
    // Page size
    int32 page_size = 3;
}

// The usage reply with time data and an actual statistics data
message UsageReply {
    // Datetime of an usage event
    repeated string datetime = 1;
    // Usage value
    repeated string usage = 2;
    // Flag to show that we have more data to load
    bool more = 3;
}